include/deal.II/base/tensor.h

```diff
@@ -924,6 +924,35 @@ Tensor<0, dim, Number>::operator-=(const Tensor<0, dim, OtherNumber> &p)
 }
 
 
+
+#ifdef __CUDA_ARCH__
+namespace internal
+{
+  template <typename Number, typename OtherNumber>
+  DEAL_II_CUDA_HOST_DEV void
+  multiply_assign_scalar(Number &tensor, const OtherNumber &s)
+  {
+    tensor *= s;
+  }
+
+  template <typename Number, typename OtherNumber>
+  DEAL_II_CUDA_HOST_DEV void
+  multiply_assign_scalar(std::complex<Number> &, const OtherNumber &)
+  {
+    printf("This function is not implemented for std::complex<Number>!\n");
+    assert(false);
```

Comment:
Couldn't you do `Assert (false, ExcMessage(...))` here?

Reply:
No, we call `host` functions in `Assert`. We use the exact same workaround in `lac/cuda_kernels.h`.